include/DummyReader.h

```diff
@@ -53,6 +53,12 @@ namespace openshot
 	 * A dummy reader can be created with any framerate or samplerate. This is useful in unit
 	 * tests that need to test different framerates or samplerates.
 	 *
+	 * @note Timeline does buffering by requesting more frames than it
+	 * strictly needs. Thus if you use this DummyReader with a custom
+	 * cache in a Timeline, make sure it has enough
+	 * frames. Specifically you need some frames after the last frame
+	 * you plan to access through the Timeline.
+	 *
```

Comment:
Makes sense. The exact number is runtime-dependent, basically it'll be OPEN_MP_NUM_PROCESSORS which is set to the smaller of:

* `num_procs()` (the actual CPU core count of the system)
* `openshot::Settings::Instance()->OMP_THREADS` (as long as that value is ≥ 2)

So, by setting `OMP_THREADS` the caller can limit the required readahead cache size to just 1 extra frame.

It might be good to add a more in-depth discussion of how all that works to the API docs for either the Settings or Timeline class, and link to that from here.

([openshot::Settings](http://openshot.org/files/libopenshot/classopenshot_1_1Settings.html#a24a29a1ca3dfaaf529f119b8fe0660c6) could really stand to be better documented in general. It doesn't help that a lot of the settings are only ever used in `#define` macros, which means they don't get indexed by Doxygen and _appear_ to be completely unreferenced in the code.)

But for the time being, this is definitely an improvement. Thanks!